sdk/storage/Azure.Storage.Blobs.Cryptography/src/EncryptedBlobClient.cs

```diff
@@ -0,0 +1,323 @@
+        }
+
+        private (Stream, EncryptionData) EncryptStream(Stream plaintext)
+        {
+            var generatedKey = CreateKey(EncryptionConstants.ENCRYPTION_KEY_SIZE_BITS);
+
+            using (AesCryptoServiceProvider aesProvider = new AesCryptoServiceProvider() { Key = generatedKey })
+            {
+
+                var encryptionData = new EncryptionData()
+                {
+                    EncryptionMode = EncryptionConstants.ENCRYPTION_MODE,
+                    ContentEncryptionIV = aesProvider.IV,
+                    EncryptionAgent = new EncryptionAgent()
+                    {
+                        EncryptionAlgorithm = ClientsideEncryptionAlgorithm.AES_CBC_256,
+                        Protocol = EncryptionConstants.ENCRYPTION_PROTOCOL_V1
+                    },
+                    KeyWrappingMetadata = new Dictionary<string, string>()
+                    {
+                        { EncryptionConstants.AGENT_METADATA_KEY, EncryptionConstants.AGENT_METADATA_VALUE }
+                    },
+                    WrappedContentKey = new WrappedKey()
+                    {
+                        Algorithm = null, // TODO
+                        EncryptedKey = this.KeyWrapper.WrapKey(KeyWrapAlgorithm, generatedKey).ToArray(),
+                        KeyId = this.KeyWrapper.KeyId
+                    }
+                };
+
+                return (
+                    new RollingBufferStream(
+                        new CryptoStream(plaintext, aesProvider.CreateEncryptor(), CryptoStreamMode.Read),
+                        EncryptionConstants.DEFAULT_ROLLING_BUFFER_SIZE,
+                        plaintext.Length + (EncryptionConstants.ENCRYPTION_BLOCK_SIZE - plaintext.Length % EncryptionConstants.ENCRYPTION_BLOCK_SIZE)),
+                    encryptionData);
+            }
+        }
+
+        /// <summary>
+        /// Securely generate a key.
+        /// </summary>
+        /// <param name="numBits">Key size.</param>
+        /// <returns>The generated key bytes.</returns>
+        private static byte[] CreateKey(int numBits)
+        {
+            using (var rng = new RNGCryptoServiceProvider())
+            {
+                var buff = new byte[numBits / 8];
+                rng.GetBytes(buff);
+                return buff;
+            }
+        }
+    }
+
+#pragma warning disable SA1402 // File may only contain a single type
+    internal static class BlobClientOptionsExtensions
+#pragma warning restore SA1402 // File may only contain a single type
+    {
+        public static BlobClientOptions WithPolicy(
+           this BlobClientOptions options,
+           HttpPipelinePolicy policy,
+           HttpPipelinePosition position = HttpPipelinePosition.PerCall)
+        {
+            if (options == default)
+            {
+                options = new BlobClientOptions();
+            }
+            options.AddPolicy(policy, position);
+            return options;
```

Comment:
We shouldn't modify the user's ClientOptions this way.  We should pass this to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/StorageClientOptions.cs#L87 (augmenting it to take another policy).  You'd then need to call the base .ctor that takes a URI and a pipeline.